weave/tests/test_op2.py

```diff
@@ -0,0 +1,178 @@
+import inspect
+from copy import deepcopy
+
+import pytest
+
+import weave
+from weave.trace import errors
+from weave.trace.op import Op, op
+from weave.trace.refs import ObjectRef
+from weave.weave_client import Call
+
+
+@op
+def func(a: int) -> int:
+    return a + 1
+
+
+@op
+async def afunc(a: int) -> int:
+    return a + 1
+
+
+class A(weave.Object):
+    @op
+    def method(self, a: int) -> int:
+        return a + 1
+
+    @op
+    async def amethod(self, a: int) -> int:
+        return a + 1
+
+
+class B:
+    """
+    special funcs (b.method.call, b.method.calls, etc.)
+    wont work as expected because it's not a weave.Object
+    """
+
+    @op
+    def method(self, a: int) -> int:
+        return a + 1
+
+    @op
+    async def amethod(self, a: int) -> int:
+        return a + 1
+
+
+a = A()
+b = B()
+
+
+def test_sync_func(client):
```

Comment:
It would be really nice if all these tests also had a version where the user publishes the object / op first, then "gets" the op and uses it.